src/scheduler.cc

```diff
@@ -296,12 +296,15 @@ Status SchedulerService::ExportFunction(ServerContext* context, const ExportFunc
   for (size_t i = 0; i < workers->size(); ++i) {
     ClientContext import_context;
     ImportFunctionRequest import_request;
-    import_request.mutable_function()->set_implementation(request->function().implementation());
+    import_request.mutable_function()->CopyFrom(request->function());
     if ((*workers)[i].current_task != ROOT_OPERATION) {
       ImportFunctionReply import_reply;
       (*workers)[i].worker_stub->ImportFunction(&import_context, import_request, &import_reply);
     }
   }
+  // TODO(rkn): Does this do a deep copy?
+  std::unique_ptr<Function> function(new Function(request->function()));
+  exported_functions.emplace_back(std::move(function));
```

Comment:
I'd just write this as

	exported_functions.push_back(std::unique_ptr<Function>(new Function(request->function())));

Also, as a general rule (not just for this project), don't use `emplace_back` unless you really need to (which should never really happen in any cases we care about). It's slightly dangerous since e.g. the previous line would work if I removed `std::unique_ptr<Function>`, but if `emplace_back` fails in that case, then the `new Function` is leaked.